lib/Extension/LanguageServerCodeTransform/CodeAction/GenerateDecoratorProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerCodeTransform\CodeAction;
+
+use Amp\CancellationToken;
+use Amp\Promise;
+use Phpactor\Extension\LanguageServerCodeTransform\LspCommand\GenerateDecoratorCommand;
+use Phpactor\LanguageServerProtocol\CodeAction;
+use Phpactor\LanguageServerProtocol\Command;
+use Phpactor\LanguageServerProtocol\Range;
+use Phpactor\LanguageServerProtocol\TextDocumentItem;
+use Phpactor\LanguageServer\Core\CodeAction\CodeActionProvider;
+use function Amp\call;
+
+class GenerateDecoratorProvider implements CodeActionProvider
+{
+    public const KIND = 'quickfix.generate_decorator';
+
+    public function kinds(): array
+    {
+        return [
+            self::KIND,
+        ];
+    }
+
+    public function provideActionsFor(TextDocumentItem $textDocument, Range $range, CancellationToken $cancel): Promise
+    {
+        return call(function () use ($textDocument) {
+            $this->generateDecoration->transform($textDocument);
+
+            return [
+                CodeAction::fromArray([
+                    'title' => 'Generate decorator',
```

Comment:
You could have `Decorate "<interface name here>"` instead of `Generate decorator

Reply:
Changed that as well.